### PR TITLE
[FIX JENKINS-45954] improvements over email attribute support

### PR DESCRIPTION
**Pull request by kuisathaverat**
[JENKINS-45954](https://issues.jenkins-ci.org/browse/JENKINS-45954)

* Check that the attribute email exists
* Check that the selected email value is not empty (null or "")
* Warning about blank email values
* possibility to log the SAMLResponse XML

@reviewbybees

### CODE REVIEW

File: src/main/java/org/jenkinsci/plugins/saml/SamlSecurityRealm.java [hunk 1]
```diff
@@ -342,18 +350,30 @@ private boolean modifyUserFullName(User user, SAML2Profile saml2Profile) {
...
-            userEmail = (String) emails.get(0);
+        if (emails == null || emails.isEmpty()) {
+            LOG.warning("There is not Email attribute '" + getEmailAttributeName() + "' for user : " + user.getId());
+            return saveUser;
+        }
+
+        for (String item : emails) {
+            if (StringUtils.isNotEmpty(item)) {
+                userEmail = item;
+                break;
```
Comment (amuniz) [inline src/main/java/org/jenkinsci/plugins/saml/SamlSecurityRealm.java]: It takes the first valid one. It should be documented in the Javadoc I think.

File: src/main/java/org/jenkinsci/plugins/saml/SamlSecurityRealm.java [hunk 2]
```diff
@@ -342,18 +350,30 @@ private boolean modifyUserFullName(User user, SAML2Profile saml2Profile) {
...
+        if (StringUtils.isBlank(userEmail)) {
+            LOG.warning("The Email is blank for user : " + user.getId());
         }
 
         try {
             if (user != null && StringUtils.isNotBlank(userEmail)) {
                 UserProperty currentUserEmailProperty = user.getProperty(UserProperty.class);
-                if (currentUserEmailProperty != null
-                        && userEmail.compareTo(StringUtils.defaultIfBlank(currentUserEmailProperty.getAddress(), "")) != 0) {
+                if (currentUserEmailProperty == null ||
```
Comment (amuniz) [inline src/main/java/org/jenkinsci/plugins/saml/SamlSecurityRealm.java]: This is changing the behavior, now the property will be set even if `currentUserEmailProperty` is null, is that what you want to achieve?

File: src/main/java/org/jenkinsci/plugins/saml/SamlSecurityRealm.java (hunk 1 above)
Comment (kuisathaverat) [inline src/main/java/org/jenkinsci/plugins/saml/SamlSecurityRealm.java]: ok

File: src/main/java/org/jenkinsci/plugins/saml/SamlSecurityRealm.java (hunk 2 above)
Comment (kuisathaverat) [inline src/main/java/org/jenkinsci/plugins/saml/SamlSecurityRealm.java]: yes, if the currentUserEmailProperty is null should create it too

**Pull request merged**